src/server/Agent.js

```diff
@@ -61,17 +65,26 @@ export default class Agent extends SideAgent {
     _startListening() {
         this.server = asyncNet.createServer();
 
-        this.server.on('connection', userSocket => this._startServing(userSocket));
+        this.server.on('connection', userSocket =>  {
+            if (!this._serverSocket) {
+                this._serverSocket = userSocket;
+                this._startServing();
+                this._serverClosed = this.server.closeAsync();
```

Comment:
Just to confirm a rather unusual intent:

```suggestion
                // we only want to server a single connection because ...
                this._serverClosed = this.server.closeAsync();
```